Generalize `active_values` (#495)
emdgroup/baybe: docs/userguide/parameters.md

```

## Discrete Parameters
A discrete parameter has a finite set of possible values that can be recommended. 
These values can be numeric or label-like (i.e. strings) and are transformed 
internally before being ingested by the surrogate model.

```{admonition} Parameter Encoding
:class: note
To make labels usable in machine learning, we assign each label one or more numbers. 
This process is called ``encoding``. While there are trivial ways of doing this 
(e.g. one-hot encoding), BayBE also provides methods to avoid problematic biases and 
even introduce useful information into the resulting latent number space. For different 
parameters, different types of encoding make sense. These situations are reflected by 
the different discrete parameter types BayBE offers.
```

```{admonition} For Label-Like Parameters: Values and Active Values
:class: note

> After a discrete parameter was created, its possible values are stored in the `values` 

Reviewer: * add link for `values`
* I think the phrase `stored in a property` is a bit confusing in the Python context. Maybe rather something like `accessible via the ... property`?

Author: added the links in the admonition

tbh I dont think a link to values etc is too helpful, they are abstract and the weird mix between private attributes and properties imo confused the reader more than it helps. Or perhaps you had a different link target in mind here?

Author: @AdrianSosic after having discussed the scaling naming thing, this is the only open item

Reviewer: Thx. From my side you can keep as is, but if you think the links should be different/not there, feel free to change. Minor anyway. Closing so that you can merge

property. All label-like parameters also support specifying `active_values`, which 

> must be ba susbet of `values`. `active_values` specify which labels are considered 

Reviewer: ```suggestion
must be a subset of `values`. `active_values` specify which labels are considered 
```

> for recommendation. Labels which are in `values` but not in `active_values` can be 

Reviewer: ```suggestion
for recommendation. Labels that are in `values` but not in `active_values` can be 
```

> part of measurements, but wont be recommended. Using this mechanism, you can configure 

Reviewer: ```suggestion
part of measurements, but won't be recommended. Using this mechanism, you can configure 
```
